kafka/consumer/fetcher.py

```diff
@@ -740,12 +740,13 @@ def _handle_fetch_response(self, request, send_time, response):
                     self._client.cluster.request_update()
                 elif error_type is Errors.OffsetOutOfRangeError:
                     fetch_offset = fetch_offsets[tp]
+                    log.info("Fetch offset %s is out of range", fetch_offset)
                     if self._subscriptions.has_default_offset_reset_policy():
                         self._subscriptions.need_offset_reset(tp)
+                        log.info("Resetting offset")
                     else:
                         self._offset_out_of_range_partitions[tp] = fetch_offset
-                    log.info("Fetch offset %s is out of range, resetting offset",
-                             fetch_offset)
+                        log.info("Raising exception")
```

Comment:
There are edge cases where we dont actually raise an exception -- if the partition gets marked unfetchable, or if a seek invalidates the fetch position.